node/pkg/reporter/app.go

```diff
@@ -175,3 +172,17 @@ func groupConfigsBySubmitIntervals(reporterConfigs []Config) map[int][]Config {
 	}
 	return grouped
 }
+
+func (a *App) handleWsMessage(ctx context.Context, data map[string]interface{}) error {
+	a.LatestData.Store(data["symbol"], data)
+	return nil
+}
+
+func (a *App) GetReporterWithInterval(interval int) (*Reporter, error) {
+	for _, reporter := range a.Reporters {
+		if reporter.SubmissionInterval == time.Duration(interval)*time.Millisecond {
+			return reporter, nil
+		}
+	}
+	return nil, errorSentinel.ErrReporterNotFound
+}
```

Comment:
it's just optional suggestion, I personally think it'd be nicer if we can store reporters as a map instead of list so that this function is unnecessary. so the key can be interval and the value can be reporter

Reply:
good suggestion! i didn't have a chance to work with this function yet because it's referenced from deviation reporter. I'll implement your suggestion in the next PR (part 5) where I implement deviation reporter